File: frontend/modelo/lineaDocVenta.ts
import { DocVenta } from "./docVenta";
import { Producto } from "./producto";
import { LineaCarrito } from "./lineaCarrito";

export interface LineaDocVenta {
    id: number;
    docVenta: DocVenta;
    producto: Producto;
    cantidad: number;
    subTotal: number;
    //tal vez no vaya, no estoy seguro
    lineaCarrito: LineaCarrito;
};

File: backend/src/main/java/pe/com/jeress/modelo/LineaDocVenta.java
package pe.com.jeress.modelo;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "LINEADOCVENTA")
@Data
public class LineaDocVenta {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "idDocVenta")
    @JsonBackReference("docventa-lineas")
    private DocVenta docVenta;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "idProducto")
    //@JsonBackReference
    @JsonIgnore
    private Producto producto;

    @Column(name="cantidad")
    private int cantidad;

    @Column(name="subTotal")
    private double subTotal;

    @OneToOne
    @JoinColumn(name = "idLineaCarrito")
    @JsonIgnore
    private LineaCarrito lineaCarrito;
}
